packages/tooling/fast-tooling-react/src/form/custom-controls/css/color-picker-component/color-picker.ts

```diff
@@ -0,0 +1,400 @@
+            {
+                this.currentRGBColor = new ColorRGBA64(
+                    param == 'r' ? newVal / 255 : this.currentRGBColor.r,
+                    param == 'g' ? newVal / 255 : this.currentRGBColor.g,
+                    param == 'b' ? newVal / 255 : this.currentRGBColor.b,
+                    param == 'a' ? newVal / 100 : this.currentRGBColor.a
+                );
+                this.currentHSVColor = rgbToHSV(this.currentRGBColor);
+                this.updateUIValues(true);
+            }
+        }
+        else if(['h','s','v'].includes(param))
+        {
+            if( (param != 'h' && this.isValideSatVal(newVal)) || ( param == 'h' && this.isValideHue(newVal) ))
+            {
+                this.updateHSV(
+                    param == 'h' ? newVal : this.currentHSVColor.h,
+                    param == 's' ? newVal / 100 : this.currentHSVColor.s,
+                    param == 'v' ? newVal / 100 : this.currentHSVColor.v
+                );
+            }
+        }
+    }
+
+    /**
+     * Change event handler for inner control.
+     * @remarks
+     * "Change" events are not `composable` so they will not
+     * permeate the shadow DOM boundary. This fn effectively proxies
+     * the change event, emitting a `change` event whenever the internal
+     * control emits a `change` event
+     * @internal
+     */
+    public handleChange(): void 
+    {
+        this.$emit("change");
+    }
+
+    /**
+     * Determines if a number value is within the valid range for an R, G, or B color channel.
+     * @param val - Number to be evaluated.
+     */
+    private isValideRGB(val: number): boolean
+    {
+        return val >= 0 && val <= 255;
+    }
+
+    /**
+     * Determines if a number value is within the valid range for the alpha channel.
+     * @param val - Number to be evaluated.
+     */
+    private isValideAlpha(val: number): boolean
+    {
+        return val >= 0 && val <= 100;
+    }
+
+    /**
+     * Determines if a number value is within the valid range for the saturation or value color channels.
+     * @param val - Number to be evaluated.
+     */
+    private isValideSatVal(val: number): boolean
+    {
+        return val >= 0 && val <= 100;
+    }
+
+    /**
+     * Determines if a number value is within the valid range for the hue color channel.
+     * @param val - Number to be evaluated.
+     */
+    private isValideHue(val: number): boolean
+    {
+        return val >= 0 && val <= 359;
+    }
+
+    /**
+     * Checks if input is a valid CSS color.
+     * After placing an invalid css color value into a color style property the value will be an empty string when read back.
+     * @internal
+     */
+    private isValideCSSColor(testValue:string): boolean 
+    {
+        /* Set the background color of the proxy element since it is not visible in the UI. */
+        this.proxy.style.backgroundColor = "";
+        this.proxy.style.backgroundColor = testValue;
```

Comment:
Can we remove the setting to an empty string here since testValue is always a required arg?

Reply:
Unfortunately no. If the color had previously been set to a valid value then it is not overwritten if the new testValue is invalid. So we have to clear it before we can use it to test again.